examples/lm1b/train.py

```diff
@@ -324,32 +326,46 @@ def tohost(x):
   return np.array(x).reshape((n_device * n_batch,) + tuple(remaining_dims))
 
 
-def main(argv):
-  if len(argv) > 1:
-    raise app.UsageError('Too many command-line arguments.')
-
+def train_and_evaluate(
+  random_seed, batch_size, learning_rate, num_train_steps, num_eval_steps,
+  eval_freq, max_target_length, max_eval_target_length, weight_decay, data_dir,
+  model_dir, restore_checkpoints, save_checkpoints, checkpoint_freq,
+  max_predict_token_length, sampling_temperature, sampling_top_k, prompt_str):
+  """Executes model training and evaluation loop.
+  
+  Args:
```

Comment:
Could we instead use the newly open-sourced `ml_collections.ConfigDict` ?

Reply:
@mohitreddy1996 has been refactoring all of our examples, enabling unit tests. My vote is to decouple this process with the adoption of ConfigDict. (And then refactoring the newly written unit tests becomes in itself part of the ConfigDict-adoption change)